test/unified_format.py

```diff
@@ -395,13 +401,18 @@ def failed(self, event):
         else:
             self.add_event(event)
 
-    def opened(self, event: ServerOpeningEvent) -> None:
+    def opened(self, event: ServerOpeningEvent | TopologyOpenedEvent) -> None:
+        self.add_event(event)
+
+    def description_changed(
+        self, event: ServerDescriptionChangedEvent | TopologyDescriptionChangedEvent
+    ) -> None:
         self.add_event(event)
 
-    def description_changed(self, event: ServerDescriptionChangedEvent) -> None:
+    def topology_changed(self, event: TopologyDescriptionChangedEvent) -> None:
         self.add_event(event)
 
-    def closed(self, event: ServerClosedEvent) -> None:
+    def closed(self, event: ServerClosedEvent | TopologyClosedEvent) -> None:
```

Comment:
This will all need to use `Union`. iirc, supporting this operator in typing is only [supported in python 3.10+](https://docs.python.org/3/library/typing.html#typing.Union)